automation-api/lib/pilot/helpers.py

```diff
@@ -0,0 +1,263 @@
+    correctness_map = {1: "correct", 2: "wrong", 3: "very wrong"}
+    # some times the model will return 'A.' instead of 'A'
+    if len(answer) == 2 and answer[1] == ".":
+        answer = answer[0]
+
+    for opt in question[1]:
+        if answer == opt.letter or answer == option_text(opt, letter_and_text=True):
+            return correctness_map[opt.correctness_of_answer_option]
+    return "failed"
+
+
+def create_question_data_for_test(question: QuestionAndOptions) -> Dict[str, str]:
+    q, options = question
+    question_dict = {"question": q.published_version_of_question}
+    question_dict["options"] = "\n".join(
+        [option_text(opt, letter_and_text=True) for opt in options]
+    )
+    return question_dict
+
+
+def create_question_dataset_for_test(
+    question_list: List[QuestionAndOptions],
+) -> List[Dict[str, str]]:
+    return [create_question_data_for_test(q) for q in question_list]
+
+
+def create_question_data_for_eval(question: QuestionAndOptions) -> Dict[str, str]:
+    q, options = question
+    question_dict = {}
+    question_text_list = [q.published_version_of_question]
+    for opt in options:
+        opt_text = option_text(opt, letter_and_text=True)
+        question_text_list.append(opt_text)
+        if opt.correctness_of_answer_option == 1:
+            question_dict["correct_answer"] = opt_text
+        elif opt.correctness_of_answer_option == 2:
+            question_dict["wrong_answer"] = opt_text
+        else:
+            question_dict["very_wrong_answer"] = opt_text
+    question_dict["question"] = "\n".join(question_text_list)
+    return question_dict
+
+
+def create_question_dataset_for_eval(
+    question_list: List[QuestionAndOptions],
+) -> List[Dict[str, str]]:
+    return [create_question_data_for_eval(q) for q in question_list]
+
+
+def check_llm_eval_output(eval_output: str) -> str:
+    eval_output = eval_output.strip().replace(".", "").lower()
+    if eval_output == "1":
+        return "correct"
+    elif eval_output == "2":
+        return "wrong"
+    elif eval_output == "3":
+        return "very wrong"
+    else:
+        return "failed"
+
+
+def get_prompt_variants(sheet: AiEvalData) -> List[PromptVariation]:
+    prompt_variations = filter_included_rows(sheet.prompt_variations.data.df)
+    res = class_objects_from_df(prompt_variations, PromptVariation)
+    return res
+
+
+def get_model_configs(sheet: AiEvalData) -> List[ModelAndConfig]:
+    models_df = sheet.gen_ai_models.data.df
+    model_configs_df = filter_included_rows(sheet.gen_ai_model_configs.data.df)
+
+    model_configs = class_objects_from_df(model_configs_df, GenAiModelConfig)
+    result = []
+    for mc in model_configs:
+        model_df = models_df.loc[models_df["model_id"] == mc.model_id]
+        model = class_objects_from_df(model_df, GenAiModel)[0]
+        result.append((model, mc))
+    return result
+
+
+def load_model_parameters(s: str) -> Dict[str, Any]:
+    if s == "nan":
```

Comment:
If a column dtype is `str` and a cell in the column is empty, then the reader will return `'nan'` for this cell. Can we just use `''` ?

Reply:
There are some higher level tweaks we can do to avoid this in the first place